testing/sdk_upgrade.py

```diff
@@ -215,8 +215,14 @@ def _wait_for_deployment(
         log.info("No config change detected. Tasks should not be restarted")
         sdk_tasks.check_tasks_not_updated(service_name, "", task_ids)
     else:
-        log.info("Checking that all tasks have restarted")
-        sdk_tasks.check_tasks_updated(service_name, "", task_ids)
+        deployment_is_complete = sdk_plan.get_deployment_plan(service_name)["status"] == "COMPLETE"
+        recovery_is_complete = sdk_plan.get_recovery_plan(service_name)["status"] == "COMPLETE"
+
+        # A difference between configs might be trivial that the scheduler doesn't consider the deployment
```

Comment:
this may still be an issue. if we have tasks hello-0 world-0 and world-1 and we get an update that only affects the hello pod, only hello-0 will be relaunched, while this code would expect world-0 and world-1 to be relaunched as well leading to a timeout